### PR TITLE
Adding additional debug text to editortest exceptions

### PR DESCRIPTION
**Pull request by evanchia-ly-sdets**
Signed-off-by: Chiang <evanchia@amazon.com>

## What does this PR do?
Adds additional debug text to EditorTest exceptions

## How was this PR tested?
No functional changes, testing in AR.

### CODE REVIEW

File: Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py [hunk 1]
```diff
@@ -1004,10 +1007,11 @@ def _run_parallel_tests(self,
                 def make_parallel_test_func(test_spec, index, current_executable):
                     def run(request, workspace, extra_cmdline_args):
                         results = self._exec_single_test(
-                            request, workspace, current_executable, index + 1, self.log_name, test_spec, extra_cmdline_args)
-                        if results is None:
-                            raise EditorToolsFrameworkException(f"Results were None. Current log name is "
-                                                                f"{self.log_name} and test is {str(test_spec)}")
+                            request, workspace, current_executable, index + 1, self.log_name, test_spec,
+                            extra_cmdline_args)
+                        if not results:
```
Comment (michleza) [inline Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py:1012]: Is this a style we should be moving to? I almost always do `if var is None:`

Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py:1012]: if X is None is fine, I just changed this to have the extra benefit of checking if the results dict is empty

**Review by AMZN-scspaldi — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

File: Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py [hunk 2]
```diff
@@ -1201,8 +1205,11 @@ def _exec_multitest(self,
                 # If it didn't then it will have "Unknown" as the type of result.
                 results = self._get_results_using_output(test_spec_list, output, executable_log_content)
                 if not len(results) == len(test_spec_list):
-                    raise EditorToolsFrameworkException("bug in get_results_using_output(), the number of results "
-                                                        "don't match the tests ran")
+                    logger.debug("\n=============Debug===========\n"
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py:1208]: IMO a debug log shouldn't need big headings like this, but I'll have to see it in action to know whether it's useful or frustrating.

**Review by Kadino — APPROVED** [review]
These look like useful improvements.

File: Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py (hunk 2 above)
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py]: removed header

**Review by jromnoa — APPROVED** [review]
Awesome changes @evanchia-ly-sdets

**Pull request merged**